gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsJdbcMonitor.java

```diff
@@ -78,8 +78,8 @@ public ClusterStats monitor(ProxyBackendConfiguration backend)
             properties.setProperty("SSL", String.valueOf(parsedUrl.getProtocol().equals("https")));
         }
         catch (MalformedURLException e) {
-            log.error("could not parse backend url %s ", url);
-            return clusterStats.build(); // TODO Invalid configuration should fail
+            log.error("Invalid backend URL configuration: %s", url);
+            throw new IllegalArgumentException("Invalid backend URL: " + url, e);
```

Comment:
We avoid both logging and throwing exceptions - either one should be sufficient.